ext/pdo_firebird/firebird_driver.c

```diff
@@ -1437,9 +1437,21 @@ static int pdo_firebird_handle_factory(pdo_dbh_t *dbh, zval *driver_options) /*
 	}
 
 	do {
+#if FB_API_VER >= 40
+		zend_string *session_timezone = pdo_attr_strval(driver_options, PDO_FB_SESSION_TIMEZONE, NULL);
+#endif
+
 		static char const dpb_flags[] = {
-			isc_dpb_user_name, isc_dpb_password, isc_dpb_lc_ctype, isc_dpb_sql_role_name };
-		char const *dpb_values[] = { dbh->username, dbh->password, vars[1].optval, vars[2].optval };
+			isc_dpb_user_name, isc_dpb_password, isc_dpb_lc_ctype, isc_dpb_sql_role_name
+#if FB_API_VER >= 40
+			, isc_dpb_session_time_zone
+#endif
+		};
+		char const *dpb_values[] = { dbh->username, dbh->password, vars[1].optval, vars[2].optval
+#if FB_API_VER >= 40
+			, session_timezone ? ZSTR_VAL(session_timezone) : NULL
+#endif
+		};
```

Comment:
Hm, I don't like this `#if` style, but I haven't a better idea.

Reply:
As I said above, I plan to add another connection parameter. So the conditional compilation directive will cover more code.